ndindex/ndindex.py

```diff
@@ -546,6 +547,164 @@ def broadcast_arrays(self):
+    skip_axes=(-1, -2))`:
+
+    >>> from ndindex import iter_indices
+    >>> for idx in iter_indices((3, 2, 4, 4), skip_axes=(-1, -2)):
+    ...     print(idx)
+    (Tuple(0, 0, slice(None, None, None), slice(None, None, None)),)
+    (Tuple(0, 1, slice(None, None, None), slice(None, None, None)),)
+    (Tuple(1, 0, slice(None, None, None), slice(None, None, None)),)
+    (Tuple(1, 1, slice(None, None, None), slice(None, None, None)),)
+    (Tuple(2, 0, slice(None, None, None), slice(None, None, None)),)
+    (Tuple(2, 1, slice(None, None, None), slice(None, None, None)),)
+
+    Note that the iterates of `iter_indices` are always a tuple, even if only
+    a single shape is provided (one could instead use `for idx, in
+    iter_indices(...)` above).
+
+    As another example, say `a` is shape `(1, 3)` and `b` is shape `(2, 1)`.
+    And you want to generate indices for every value of the broadcasted
+    operation `a + b`. You could use `a[idx1.raw] + b[idx2.raw]` for every
+    `idx1` and `idx2` as below:
+
+    >>> import numpy as np
+    >>> a = np.arange(3).reshape((1, 3))
+    >>> b = np.arange(100, 111, 10).reshape((2, 1))
+    >>> a
+    array([[0, 1, 2]])
+    >>> b
+    array([[100],
+           [110]])
+    >>> for idx1, idx2 in iter_indices((1, 3), (2, 1)):
+    ...     print(f"{idx1 = }; {idx2 = }; {(a[idx1.raw], b[idx2.raw]) = }")
+    idx1 = Tuple(0, 0); idx2 = Tuple(0, 0); (a[idx1.raw], b[idx2.raw]) = (0, 100)
+    idx1 = Tuple(0, 1); idx2 = Tuple(0, 0); (a[idx1.raw], b[idx2.raw]) = (1, 100)
+    idx1 = Tuple(0, 2); idx2 = Tuple(0, 0); (a[idx1.raw], b[idx2.raw]) = (2, 100)
+    idx1 = Tuple(0, 0); idx2 = Tuple(1, 0); (a[idx1.raw], b[idx2.raw]) = (0, 110)
+    idx1 = Tuple(0, 1); idx2 = Tuple(1, 0); (a[idx1.raw], b[idx2.raw]) = (1, 110)
+    idx1 = Tuple(0, 2); idx2 = Tuple(1, 0); (a[idx1.raw], b[idx2.raw]) = (2, 110)
+    >>> a + b
+    array([[100, 101, 102],
+           [110, 111, 112]])
+
+    To include an index into the final broadcasted array, you can simply
+    include the final broadcasted shape as one of the shapes (the NumPy
+    function :func:`np.broadcast_shapes <numpy:numpy.broadcast_shapes>` is useful here).
+
+    >>> np.broadcast_shapes((1, 3), (2, 1))
+    (2, 3)
+    >>> for idx1, idx2, broadcasted_idx in iter_indices((1, 3), (2, 1), (2, 3)):
+    ...     print(broadcasted_idx)
+    Tuple(0, 0)
+    Tuple(0, 1)
+    Tuple(0, 2)
+    Tuple(1, 0)
+    Tuple(1, 1)
+    Tuple(1, 2)
+
+    """
+    if not shapes:
+        yield ()
+        return
+
+    shapes = [asshape(shape) for shape in shapes]
+    ndim = len(max(shapes, key=len))
+
+    if isinstance(skip_axes, int):
+        skip_axes = (skip_axes,)
+    _skip_axes = []
+    for a in skip_axes:
+        try:
+            a = ndindex(a).reduce(ndim).args[0]
+        except IndexError:
+            # Raise the same error as NumPy functions that take axis arguments
+            raise AxisError(f"axis {a} is out of bounds for array of dimension {ndim}")
```

Comment:
```suggestion
            raise AxisError(a, ndim)
```
This follows the `np.AxisError` signature, which will still generate the error message you desire. https://github.com/numpy/numpy/blob/f4353323d0be0c3b3c7d0c752d6c135bc6d36465/numpy/core/_exceptions.py#L124